mopro-ffi/src/lib.rs

```diff
@@ -80,40 +87,56 @@ pub enum MoproError {
     Halo2Error(String),
 }
 
+//
+// Circom Proof
+//
 #[derive(Debug, Clone)]
-pub struct GenerateProofResult {
-    pub proof: Vec<u8>,
-    pub inputs: Vec<u8>,
+pub struct CircomProofResult {
+    pub proof: CircomProof,
+    pub inputs: Vec<String>,
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct CircomProof {
+    pub a: G1,
+    pub b: G2,
+    pub c: G1,
+    pub protocol: String,
+    pub curve: String,
 }
 
-#[cfg(not(feature = "circom"))]
 #[derive(Debug, Clone, Default)]
 pub struct G1 {
     pub x: String,
     pub y: String,
+    pub z: Option<String>,
 }
 
-#[cfg(not(feature = "circom"))]
 #[derive(Debug, Clone, Default)]
 pub struct G2 {
     pub x: Vec<String>,
     pub y: Vec<String>,
+    pub z: Option<Vec<String>>,
 }
 
-#[cfg(not(feature = "circom"))]
-#[derive(Debug, Clone, Default)]
-pub struct ProofCalldata {
-    pub a: G1,
-    pub b: G2,
-    pub c: G1,
+//
+// Halo2 Proof
+//
+#[derive(Debug, Clone)]
+pub struct Halo2ProofResult {
+    pub proof: Vec<u8>,
+    pub inputs: Vec<u8>,
 }
 
-#[derive(Debug, Clone, Default)]
-pub enum ProofLib {
-    #[default]
-    Arkworks,
-    Rapidsnark,
-}
+//
+// Proof Libs
+//
+// #[derive(Debug, Clone, Default)]
+// pub enum ProofLib {
+//     #[default]
+//     Arkworks,
+//     Rapidsnark,
+// }
```

Comment:
this can be removed?

Reply:
Yep, forgot to remove it.
fixed in [a0551c9](https://github.com/zkmopro/mopro/pull/372/commits/a0551c983b38309cd18d6e21617883a74b541b1f)